acceptance/Rakefile

```diff
@@ -17,6 +17,73 @@ EOS
       ENV['TESTS'] = get_test_sample.join(",")
       Rake::Task["ci:test:aio"].invoke
     end
+
+    desc <<-EOS
+Run tests on docker quickly and easily. The docker container is set up to mount
+your puppet directory in the container. This means you can edit code or test
+files and rerun tests without reconfiguring your test environment.
+
+Defaults to running all tests unless TESTS is set. TESTS is a comma seperated
+list of test files to run.
+
+  $ bundle exec rake ci:test:docker TESTS='tests/puppet_apply_basics.rb,tests/resource/file/content_attribute.rb'
+
+By default, tests are run on a centos 7 host. To change the host, set HOSTS to
+a valid host string according to beaker-hostgenerator requirements.
+
+All tests marked with a server tag will be skipped.
+
+This task skips all cleanup. Please be sure to run `bundle exec beaker destroy`
+to clean up docker containers used for testing.
+EOS
+    task :docker do
+      ENV['HOSTS'] ||= 'centos7-64a'
+      ENV['SHA'] ||= `git rev-parse HEAD`.chomp
+      ENV['OPTIONS'] ||= '--preserve-hosts=always'
+      ENV['OPTIONS'] += ' --test-tag-exclude=server'
+      Rake::Task["ci:gen_hosts"].invoke('docker')
+      hosts_file_content = YAML.load_file ENV['HOSTS']
+      hosts_file_content['HOSTS'].each do |host|
+        host[1]['mount_folders'] = {
+          'puppet' => {
+            'host_path' => "#{File.dirname(__dir__)}" ,
+            'container_path' => '/build/puppet'
+          }
+        }
+        host[1]['tag'] = 'acceptance_test_host'
+      end
+      File.open(ENV['HOSTS'], "w") { |f| f.write(YAML.dump(hosts_file_content)) }
+      Rake::Task["ci:test:git"].invoke
+
+    ensure
```

Comment:
unfortunately the "implicit begin" is only supported in newer ruby version. When I run with 2.3.8 it barfs with `SyntaxError: /Users/josh/work/puppet/acceptance/Rakefile:58: syntax error, unexpected keyword_ensure, expecting keyword_end`. Could you add an explicit `begin ... ensure ... end`?

Reply:
okie, tested this with ruby 2.3.8 and it works